docs/core/extensions/dependency-injection-usage.md

```diff
@@ -30,55 +30,73 @@ In this tutorial, you learn how to:
+All of the subinterfaces of `IReportServiceLifetime` explicitly implement the `IReportServiceLifetime.Lifetime` with a default. For example, `IExampleTransientService` explicitly implements `IReportServiceLifetime.Lifetime` with the `ServiceLifetime.Transient` value.
 
-All of the subinterfaces of `IOperation` name their intended service lifetime. For example, "Transient" or "Singleton".
+## Add default implementations
 
-## Add default implementation
+The example implementations all initialize their `Id` property with the result of <xref:System.Guid.NewGuid?displayProperty=nameWithType>. Add the following default implementation classes for the various services to the project root directory:
 
-Add the following default implementation for the various operations:
+*ExampleTransientService.cs*
 
-*DefaultOperation.cs*
+:::code source="snippets/configuration/console-di/ExampleTransientService.cs":::
 
-:::code language="csharp" source="snippets/configuration/console-di/DefaultOperation.cs":::
+*ExampleScopedService.cs*
 
-The `DefaultOperation` implements all of the named marker interfaces and initializes the `OperationId` property to the last four characters of a new globally unique identifier (GUID).
+:::code source="snippets/configuration/console-di/ExampleScopedService.cs":::
 
-## Add service that requires DI
+*ExampleSingletonService.cs*
 
-Add the following operation logger object, which acts as a service to the console app:
+:::code source="snippets/configuration/console-di/ExampleSingletonService.cs":::
 
-*OperationLogger.cs*
+Each implementation is defined as `internal sealed` and implements its corresponding interface. For example, `ExampleSingletonService` implements `IExampleSingletonService`.
 
-:::code language="csharp" source="snippets/configuration/console-di/OperationLogger.cs":::
+## Add a service that requires DI
 
-The `OperationLogger` defines a constructor that requires each of the aforementioned marker interfaces, that is, `ITransientOperation`, `IScopedOperation`, and `ISingletonOperation`. The object exposes a single method that allows the consumer to log the operations with a given `scope` parameter. When invoked, the `LogOperations` method logs each operation's unique identifier with the scope string and message.
+Add the following service lifetime reporter class, which acts as a service to the console app:
+
+*ServiceLifetimeReporter.cs*
+
+:::code source="snippets/configuration/console-di/ServiceLifetimeReporter.cs":::
+
+The `ServiceLifetimeReporter` defines a constructor that requires each of the aforementioned marker interfaces, that is, `IExampleTransientService`, `IExampleScopedService`, and `IExampleSingletonService`. The object exposes a single method that allows the consumer to report on the service with a given `lifetimeDetails` parameter. When invoked, the `ReportServiceLifetimeDetails` method logs each service's unique identifier with the service lifetime message.
```

Comment:
It looks like you removed the line (65) where marker interfaces were previously mentioned, so I think some further explanation is needed.